tests/rptest/tests/tombstone_removal_e2e_test.py

```diff
@@ -0,0 +1,283 @@
+# by the Apache License, Version 2.0
+
+import time
+
+import requests
+
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.admin import Admin
+from rptest.services.kgo_verifier_services import KgoVerifierProducer
+from rptest.tests.partition_movement import PartitionMovementMixin
+from ducktape.tests.test import TestContext
+
+
+class TombstoneRemovalTest(RedpandaTest):
+    topic_name = "tr_test_topic"
+    data_msg_count = 256 * 1024
+    data_msg_size = 1024
+
+    def __init__(self, test_context: TestContext):
+        extra_rp_conf = dict(
+            enable_leader_balancer=False,
+            partition_autobalancing_mode="off",
+            health_monitor_max_metadata_age=100,  # ms
+            raft_learner_recovery_rate=10 * 1024 * 1024,  # 10MB/s
+            tombstone_retention_ms=1000,  # 1 second
+        )
+        super(TombstoneRemovalTest, self).__init__(
+            num_brokers=5,
+            test_context=test_context,
+            extra_rp_conf=extra_rp_conf,
+        )
+
+    def produce_data(self, key_set_cardinality: int):
+        producer = KgoVerifierProducer(
+            self.test_context,
+            self.redpanda,
+            self.topic_name,
+            msg_size=self.data_msg_size,
+            msg_count=self.data_msg_count,
+            key_set_cardinality=key_set_cardinality,
+            tombstone_probability=0.0,
+        )
+        try:
+            producer.start()
+            producer.wait_for_acks(self.data_msg_count, 100, 1)
+        finally:
+            producer.stop()
+            producer.free()
+
+    def leadership_transferred_to_node(self, node_id: int) -> bool:
+        current_leader = self.admin.get_partition("kafka", self.topic_name, 0)[
+            "leader_id"
+        ]
+        if current_leader == node_id:
+            return True
+        self.redpanda.logger.debug(
+            f"Current leader is {current_leader}, trying to transfer to node {node_id}"
+        )
+        try:
+            self.admin.transfer_leadership_to(
+                namespace="kafka",
+                topic=self.topic_name,
+                partition=0,
+                target_id=node_id,
+            )
+        except requests.exceptions.HTTPError as e:
+            self.redpanda.logger.debug(
+                f"Failed to transfer leadership to node {node_id}, retrying: {e}"
+            )
+        return False
+
+    def transfer_leadership_to_node(self, node_id: int):
+        wait_until(
+            lambda: self.leadership_transferred_to_node(node_id),
+            timeout_sec=30,
+            backoff_sec=1,
+            err_msg=f"timed out waiting for leadership transfer to node {node_id}",
+        )
+
+    def get_data59_from_node(self, node_id: int):
+        self.transfer_leadership_to_node(node_id)
```

Comment:
although the balancer is disabled, the leadership may still move due to network latency, right? i.e. in cdt. For a bulletproof test it needs to assert at the end that leadership/epoch didn't change in-between.

You are expecting leadership to be stable but this expectation can be broken in too many ways in the future and by new settings... (like [this](https://github.com/redpanda-data/redpanda/pull/28331) one)